src/v/kafka/client/cluster.cc

```diff
@@ -173,46 +184,106 @@ ss::future<> cluster::initialize_metadata_with_seed() {
     }
 }
 
-ss::future<> cluster::dispatch_metadata_request() {
+ss::future<> cluster::dispatch_and_apply_updates(
+  std::optional<chunked_vector<model::topic>> topic_request_list) {
     auto h = _gate.hold();
-    vlog(_logger.debug, "Dispatching metadata request");
+    vlog(
+      _logger.debug, "Dispatching and applying cluster and metadata updates");
+    std::optional<metadata_response_data> metadata_resp;
     shared_broker_t broker;
 
     if (_brokers.empty()) {
-        // If there are no brokers, connect to one of the seeds
         co_await initialize_metadata_with_seed();
     }
 
     try {
-        auto broker = _brokers.any();
-        // Initialize the broker if it is not connected, this is required to
-        // make sure the broker has up to date API version information.
-
-        auto request_version = co_await get_metadata_request_version(
-          broker, _as);
-        // TODO: support topic subscription
-        auto reply = co_await broker->dispatch(
-          metadata_request{.data{.include_topic_authorized_operations = true}},
-          request_version);
-        vassert(
-          std::holds_alternative<kafka::metadata_response>(reply),
-          "Metadata response is required to be returned as a result of "
-          "metadata request");
+        broker = _brokers.any();
+
+        auto cluster_describe_resp = co_await dispatch_describe_cluster_request(
```

Comment:
Again, here we lost support for cluster pre describe_cluster_request

Reply:
This should be done conditionally, if describe_cluster is not supported we should fallback to metadata request